dataset_reader/base_reader.py

```diff
@@ -2,20 +2,48 @@
 from typing import Iterator, List, Optional
 
 
+@dataclass
+class SparseVector:
+    indices: List[int]
+    values: List[float]
+
+
 @dataclass
 class Record:
     id: int
-    vector: List[float]
+    vector: Optional[List[float]]
+    sparse_vector: Optional[SparseVector]
     metadata: Optional[dict]
 
+    def __post__init__(self):
```

Comment:
overall it is probably a good check
however, i am not sure whether we need it in benchmarks (don't like checking something for each record, however i might be unnecessary careful here)

method name is wrong
it is __post_init__ with one underscore in the middle
the condition is also wrong

`(dense_vector or sparse_vector) and not (dense_vector and sparse_vector)`

this one checks the following:
if (at least one of vectors is set) and not (both vectors are set): raise

which is the opposite of what we want

we actually want if dense_vector == sparse_vector: raise

btw, the naming might be improved as well